test/Web/ScottySpec.hs

```diff
@@ -41,6 +47,16 @@ main = hspec spec
 availableMethods :: [StdMethod]
 availableMethods = [GET, POST, HEAD, PUT, PATCH, DELETE, OPTIONS]
 
+data SearchForm = SearchForm
+  { sfQuery :: Text
+  , sfYear :: Int
+  } deriving (Generic)
+
+instance FromForm SearchForm where
+
+postForm :: ByteString -> LBS.ByteString -> WaiSession st SResponse
```

Comment:
Ah, I did this for two reasons: Mostly because `postHtmlForm` turns regular Haskell strings into url-encoded strings before doing the request, so using this custom one makes the encoding more explicit in the tests. This makes them easier to read to me but I realise that's personal preference.

Second reason is that `postForm` was already present in this module, I just moved it to the top level for code reuse.

Happy to change it either way :+1:  Let me know :)